web/src/main/webapp/dataViews/tableView.ts

```diff
@@ -492,6 +501,140 @@ export class TableView extends TSViewBase implements IScrollTarget, OnNextK {
+                action: () => this.pca(true),
+                help: "Perform Principal Component Analysis on a set of numeric columns. " +
+                    "This produces a smaller set of columns that preserve interesting properties of the data.",
+            }, selectedCount > 1 &&
+				     this.getSelectedColNames().reduce((a, b) => a && this.isNumericColumn(b), true));
+            this.contextMenu.addItem({
+                text: "Plot Singular Value Spectrum",
+                action: () => this.spectrum(true),
+                help: "Plot singular values for the selected columns. ",
+            }, selectedCount > 1 &&
+				     this.getSelectedColNames().reduce((a, b) => a && this.isNumericColumn(b), true));
+            this.contextMenu.addItem({
+                text: "Filter...",
+                action: () => {
+                    const colName = this.getSelectedColNames()[0];
+                    const colDesc = this.schema.find(colName);
+                    this.showFilterDialog(colDesc.name, this.order, this.tableRowsDesired);
+                },
+                help: "Eliminate data that matches/does not match a specific value.",
+            }, selectedCount === 1);
+            this.contextMenu.addItem({
+                text: "Compare...",
+                action: () => {
+                    const colName = this.getSelectedColNames()[0];
+                    this.showCompareDialog(this.schema.displayName(colName),
+					   this.order, this.tableRowsDesired);
+                },
+                help: "Eliminate data that matches/does not match a specific value.",
+            }, selectedCount === 1);
+            this.contextMenu.addItem({
+                text: "Convert...",
+                action: () => this.convert(cd.name),
+                help: "Convert the data in the selected column to a different data type.",
+            }, selectedCount === 1);
+            this.contextMenu.addItem({
+                text: "Create column in JS...",
+                action: () => this.createJSColumnDialog(this.order, this.tableRowsDesired),
+                help: "Add a new column computed using Javascript from the selected columns.",
+            }, true);
+            this.contextMenu.addItem({
+                text: "Extract value...",
+                action: () => {
+                    const colName = this.getSelectedColNames()[0];
+                    this.createKVColumnDialog(colName, this.tableRowsDesired);
+                },
+                help: "Extract a value associated with a specific key.",
+            }, selectedCount === 1 &&
+				     this.isKVColumn(this.getSelectedColNames()[0]));
+        } else {
+            this.contextMenu.addItem({
+                text: "Private Histogram",
+                action: () => this.privateHistSelected(),
+                help: "Plot the data in the selected columns as a private histogram. " +
+                    "Applies to one numeric column only.",
```

Comment:
Instead of this line add a check below selectedCount === 1 && kindIsNumeric(cd.kind).
(You may also want to add something about dates too...)